src/Reflection/Adapter/ReflectionClass.php

```diff
@@ -0,0 +1,442 @@
+        foreach ($properties as $key => $property) {
+            $wrappedProperties[$key] = new ReflectionProperty($property);
+        }
+        return $wrappedProperties;
+    }
+
+    /**
+     * @param string $name
+     * @return bool
+     */
+    public function hasConstant($name)
+    {
+        return $this->betterReflectionClass->hasConstant($name);
+    }
+
+    /**
+     * @return mixed[]
+     */
+    public function getConstants()
+    {
+        return $this->betterReflectionClass->getConstants();
+    }
+
+    /**
+     * @param string $name
+     * @return mixed|null
+     */
+    public function getConstant($name)
+    {
+        return $this->betterReflectionClass->getConstant($name);
+    }
+
+    /**
+     * @return ReflectionClass[]
+     */
+    public function getInterfaces()
+    {
+        $interfaces = $this->betterReflectionClass->getInterfaces();
+
+        $wrappedInterfaces = [];
+        foreach ($interfaces as $key => $interface) {
+            $wrappedInterfaces[$key] = new ReflectionClass($interface);
+        }
+        return $wrappedInterfaces;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getInterfaceNames()
+    {
+        return $this->betterReflectionClass->getInterfaceNames();
+    }
+
+    /**
+     * @return bool
+     */
+    public function isInterface()
+    {
+        return $this->betterReflectionClass->isInterface();
+    }
+
+    /**
+     * @return ReflectionClass[]
+     */
+    public function getTraits()
+    {
+        $traits = $this->betterReflectionClass->getTraits();
+
+        $wrappedTraits = [];
+        foreach ($traits as $key => $trait) {
+            $wrappedTraits[$key] = new ReflectionClass($trait);
+        }
+        return $wrappedTraits;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getTraitNames()
+    {
+        return $this->betterReflectionClass->getTraitNames();
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getTraitAliases()
+    {
+        return $this->betterReflectionClass->getTraitAliases();
+    }
+
+    /**
+     * @return bool
+     */
+    public function isTrait()
+    {
+        return $this->betterReflectionClass->isTrait();
+    }
+
+    /**
+     * @return bool
+     */
+    public function isAbstract()
+    {
+        return $this->betterReflectionClass->isAbstract();
+    }
+
+    /**
+     * @return bool
+     */
+    public function isFinal()
+    {
+        return $this->betterReflectionClass->isFinal();
+    }
+
+    /**
+     * @return int
+     */
+    public function getModifiers()
+    {
+        return $this->betterReflectionClass->getModifiers();
+    }
+
+    /**
+     * @param object $object
+     * @return bool
+     */
+    public function isInstance($object)
+    {
+        return $this->betterReflectionClass->isInstance($object);
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function newInstance($args = null, $_ = null)
```

Comment:
Also, this is runtime reflection: you will need to rely on the actual internal reflection at this point (or rely on `doctrine/instantiator`)